client/src/controllers/RevealController.ts

```diff
@@ -0,0 +1,58 @@
+import { Controller } from '@hotwired/stimulus';
+
+/**
+ *
+ * @example
+ * <section
+ * data-controller="w-panel"
+ * data-action="w-panel:toggle->w-panel#toggle"
+ * >
+ *   <div>
+ *       <a w-panel-target="anchor">
+ *       </a>
+ *       <button data-action="w-panel->toggle">
+ *       </button>
+ *           <h2 w-panel-target="heading">
+ *                   <span w-panel-target="heading-text"></span>
+ *           </h2>
+ *           <div w-panel-target="controls">
+ *           </div>
+ *   </div>
+ *   <div
+ * w-panel-target="item"
+ * data-action="beforematch->w-panel#toggle"
+ * >
+ *   </div>
+ * </section>
+ * */
+export class RevealController extends Controller<HTMLButtonElement> {
```

Comment:
```suggestion
export class RevealController extends Controller<HTMLElement> {
```

The controlled element is actually the root div, so just HTMLElement here is sufficient.